### ISSUE TITLE
[DSv4 Support] Indexer, TopK, Indexer + TopK Fused

### ISSUE THREAD

**Issue by dhiraj113** [flashinfer-ai/flashinfer#3520]
Indexer BMM + Relu + Scaled Sum with scaling weights for each head [IDX-BMM]

GEMM -  (128, S, 128) + Fused Relu + Scaled sum across 128 heads  to produce - S index scores

TopK [IDX-TOPK]- 

 From the S index scores, obtain indices for the top 1024 scores, and return the indices.

Fusion - 

Indexer BMM with Relu and Scaled Sum fused with TopK [IDX-BMM-MEGA]

Benefit - Don’t have write and read back [S, 1]

**Comment by leonardHONG** [comment]
Hi @dhiraj113, I’d like to take the SM120 side of IDX-BMM and add a native FlashInfer paged-MQA logits provider. I’ll keep the API aligned with #4365 to avoid duplicating the SM100/SM103 work. Is anyone already working on this?

**Comment by dhiraj113** [comment]
Hi @leonardHONG, it would be great if you can contribute to the SM120 kernels/integration. I am not aware of anyone currently working on SM120 for these.
